src/main/scala/scalismo/registration/TransformationSpace.scala

```diff
@@ -618,34 +654,100 @@ object RigidTransformationSpace {
-    extends ProductTransformation[D](rotation, translation) with RigidTransformation[D] {
-  def inverse = new RigidTransformationRotThenTrans[D](translation.inverse, rotation.inverse)
+trait SimilarityTransformation[D <: Dim] extends CompositeTransformation[D] with CanInvert[D] {
+  def rigidTransformation: RigidTransformation[D]
+  def scaling: ScalingTransformation[D]
+  def inverse: SimilarityTransformation[D]
+}
+
+object SimilarityTransformation {
+  /**
+   * Constructs a similarity transformation, which first applies a rigid transformation and then
+   * a scaling transformation
+   *
+   * @param scaling
+   * @param rigidTransformation
+   * @tparam D
+   * @return
+   */
+  def apply[D <: Dim: NDSpace](scaling: ScalingTransformation[D], rigidTransformation: RigidTransformation[D]): SimilarityTransformation[D] = {
+    new SimilarityTransformationScalingFollowingRigid[D](scaling, rigidTransformation)
+  }
+  def apply[D <: Dim: NDSpace](rigidTransformation: RigidTransformation[D], scaling: ScalingTransformation[D]): SimilarityTransformation[D] = {
+    new SimilarityTransformationRigidFollowingScaling[D](rigidTransformation, scaling)
+  }
+}
+
+private class SimilarityTransformationScalingFollowingRigid[D <: Dim: NDSpace](val scaling: ScalingTransformation[D],
+  val rigidTransformation: RigidTransformation[D])
+    extends CompositeTransformation[D](scaling, rigidTransformation) with SimilarityTransformation[D] {
 
+  override def inverse: SimilarityTransformation[D] = new SimilarityTransformationRigidFollowingScaling[D](rigidTransformation.inverse, scaling.inverse)
+}
+
+private class SimilarityTransformationRigidFollowingScaling[D <: Dim: NDSpace](val rigidTransformation: RigidTransformation[D],
+  val scaling: ScalingTransformation[D])
+    extends CompositeTransformation[D](scaling, rigidTransformation) with SimilarityTransformation[D] {
+
+  override def inverse: SimilarityTransformation[D] = new SimilarityTransformationScalingFollowingRigid(scaling.inverse, rigidTransformation.inverse)
 }
 
 /**
- *  Anisotropic scaling transform, where each dimension is scaled differently
+ * Parametric transformation space producing similarity transforms.
+ *
+ * @constructor Returns a parametric space generating anisotropic similarity transforms
+ * @param center : center of rotation used in the rigid transform
+ */
+case class SimilarityTransformationSpace[D <: Dim: NDSpace: CreateRotationSpace: ScalingSpace.Create](center: Point[D])
+    extends ProductTransformationSpace[D, ScalingTransformation[D], RigidTransformation[D]](ScalingSpace[D](), RigidTransformationSpace[D](center)) {
+
+  /**
+   * Returns a D-dimensional c similarity transform that performs the rigid transofrmation first, then a scaling.
+   *
+   * @param p parameter vector for the transform. This must be a concatenation of the rigid transform parameters first, then scaling parameters
```

Comment:
The API doc description does not match the code below. When using splitProductParameterVector, you assume that the scaling parameters come first